Learning_FEM_1D_step0.ipynb

```diff
@@ -347,38 +347,41 @@
     "If we report our approximate solution in the equation, we get:\n",
     "\n",
     "$$ -\\frac{d}{dx}D\\frac{du_h}{dx} + \\sigma u_h \\ne q \\qquad \\forall x \\in [0,L]$$\n",
+    "(As $u_h$ is an approximation to our solution, we do not expect the equality to hold for every $x$. There should exist some residual function $R(u_h)$ defined as:\n",
+    "$$R(u_h):=q-(-\\frac{d}{dx}D\\frac{du_h}{dx}+\\sigma u_h)$$\n",
+    "That is why the inequality.)\n",
     "\n",
     "This is not very useful. First, our approximate solution cannot be differentiated twice in $x$. Second, we need to obtain equations **for all** of the unknown nodal values $u_j$ with $0 \\le j \\le N$.\n",
     "\n",
     "One way to solve these issues is to integrate that equation over the entire domain (getting rid of the $x$ variable)\n",
     "\n",
-    "$$ \\int_0^L dx \\left( -\\frac{d}{dx}D\\frac{du_h}{dx} + \\sigma u_h = q \\right)$$\n",
+    "$$ \\int_0^L dx \\left( -\\frac{d}{dx}D\\frac{du_h}{dx} + \\sigma u_h \\ne q \\right)$$\n",
```

Comment:
this is not correct. 

We seek a weak formulation where the residual, integrated over space, is 0.

Reply:
Corrected as per suggestion